girder/events.py

```diff
@@ -206,12 +207,30 @@ def unbind(eventName, handlerName):
 
 def unbindAll():
     """
-    Clears the entire event map. Any bound listeners will be unbound.
+    Clears the entire event map. All bound listeners will be unbound.
+
+     .. warning:: This will also disable internal event listeners, which are
+       necessary for normal Girder functionality. This function should generally
+       never be called outside of testing.
     """
     global _mapping
     _mapping = {}
 
 
+@contextlib.contextmanager
+def bound(eventName, handlerName, handler):
+    """
+    A context manager to temporarily bind an event handler within its scope.
+
+    Parameters are the same as those to :py:func:`girder.events.bind`.
+    """
+    bind(eventName, handlerName, handler)
+    try:
+        yield
```

Comment:
No, the goal is pause on any exceptions that happen in user code, just to unbind the event.

Note that there is no ``except`` clause in this block, since all exceptions should be passed on up the call stack. If there was an exception inside ``bind`` (unlikely), there's no reason to call ``unbind`` (things would already be too screwed up at that point).